core/scripts/common/polygonedge.go

```diff
@@ -0,0 +1,188 @@
+	var h PolygonEdgeHeader
+	err = jsonRPCClient.Call(&h, "eth_getBlockByNumber", "0x"+blockNum.Text(16), true)
+	if err != nil {
+		return nil, "", fmt.Errorf("failed to get poloygon-edge header: %+v", err)
+	}
+
+	ar := &fastrlp.Arena{}
+	val, err := MarshalRLPWith(ar, &h)
+	if err != nil {
+		return nil, "", err
+	}
+
+	dst := make([]byte, 0)
+	dst = val.MarshalTo(dst)
+
+	return dst, h.Hash.String(), err
+}
+
+// MarshalRLPWith marshals the header to RLP with a specific fastrlp.Arena
+// Adding polygon-edge as a dependency caused a lot of issues with conflicting
+// dependency version with other libraries in this repo and some methods being
+// referenced from older versions
+// Reference: https://github.com/0xPolygon/polygon-edge/blob/develop/types/rlp_marshal.go#L73C50-L73C53
+func MarshalRLPWith(arena *fastrlp.Arena, h *PolygonEdgeHeader) (*fastrlp.Value, error) {
+	vv := arena.NewArray()
+
+	vv.Set(arena.NewCopyBytes(h.ParentHash.Bytes()))
+	vv.Set(arena.NewCopyBytes(h.Sha3Uncles.Bytes()))
+	vv.Set(arena.NewCopyBytes(h.Miner[:]))
+	vv.Set(arena.NewCopyBytes(h.StateRoot.Bytes()))
+	vv.Set(arena.NewCopyBytes(h.TxRoot.Bytes()))
+	vv.Set(arena.NewCopyBytes(h.ReceiptsRoot.Bytes()))
+	vv.Set(arena.NewCopyBytes(h.LogsBloom[:]))
+
+	vv.Set(arena.NewUint(uint64(h.Difficulty)))
+	vv.Set(arena.NewUint(uint64(h.Number)))
+	vv.Set(arena.NewUint(uint64(h.GasLimit)))
+	vv.Set(arena.NewUint(uint64(h.GasUsed)))
+	vv.Set(arena.NewUint(uint64(h.Timestamp)))
+
+	extraDataBytes, err := h.ExtraData.Decode()
+	if err != nil {
+		return nil, fmt.Errorf("failed to hex decode polygon-edge ExtraData: %+v", err)
+	}
+	extraDataBytes, err = GetIbftExtraClean(extraDataBytes)
+	if err != nil {
+		return nil, fmt.Errorf("GetIbftExtraClean error : %+v", err)
+	}
+	vv.Set(arena.NewCopyBytes(extraDataBytes))
+	vv.Set(arena.NewCopyBytes(h.MixHash.Bytes()))
+
+	nonceHexString := h.Nonce.String()
+	nonceBytes, err := hexutil.Decode(nonceHexString)
+	if err != nil {
+		return nil, fmt.Errorf("failed to hex decode polygon-edge ExtraData: %+v", err)
+	}
+	vv.Set(arena.NewCopyBytes(nonceBytes))
+
+	baseFee := h.BaseFee
+	if h.BaseFeeAlt > 0 {
+		baseFee = h.BaseFeeAlt
+	}
+	vv.Set(arena.NewUint(uint64(baseFee)))
+
+	return vv, nil
+}
+
+// Remove blockHeader.ExtraData.Committed without unpacking ExtraData into
+// its full fledged type, which needs the full import of the package
+// github.com/0xPolygon/polygon-edge. polygon-edge is a node implementation,
+// and not a client. Adding polygon-edge as a dependency caused a lot of
+// issues with conflicting dependency version with other libraries in this
+// repo and some methods being referenced from older versions.
+func GetIbftExtraClean(extra []byte) (cleanedExtra []byte, err error) {
+	// Capture prefix 0's sent by nexon supernet
+	hexExtra := hex.EncodeToString(extra)
+	prefix := ""
+	for _, s := range hexExtra {
+		if string(s) == string("0") {
```

Comment:
You can compare runes:
```suggestion
		if s == '0' {
```